tests/build/build.go

```diff
@@ -879,7 +889,167 @@ var _ = framework.BuildSuiteDescribe("Build service E2E tests", Label("build", "
+					Application:   applicationName,
+					Source: appservice.ComponentSource{
+						ComponentSourceUnion: appservice.ComponentSourceUnion{
+							GitSource: &appservice.GitSource{
+								URL:      secretLookupComponentTwoGitSourceURL,
+								Revision: secondComponentBaseBranchName,
+							},
+						},
+					},
+				}
+				_, err := f.AsKubeAdmin.HasController.CreateComponent(componentObj2, testNamespace, "", "", applicationName, false, utils.MergeMaps(constants.ComponentPaCRequestAnnotation, constants.ImageControllerAnnotationRequestPublicRepo))
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+
+			It("check first component annotation has errors", func() {
+				buildStatus := &controllers.BuildStatus{}
+				Eventually(func() (bool, error) {
+					component, err := f.AsKubeAdmin.HasController.GetComponent(firstComponentName, testNamespace)
+					if err != nil {
+						return false, err
+					} else if component == nil {
+						return false, fmt.Errorf("got component as nil after getting component %s in namespace %s", firstComponentName, testNamespace)
+					}
+					buildStatusAnnotationValue := component.Annotations[controllers.BuildStatusAnnotationName]
+					GinkgoWriter.Printf(buildStatusAnnotationValueLoggingFormat, buildStatusAnnotationValue)
+					statusBytes := []byte(buildStatusAnnotationValue)
+					err = json.Unmarshal(statusBytes, buildStatus)
+					if err != nil {
+						return false, err
+					}
+					return buildStatus.PaC != nil && buildStatus.PaC.State == "error" && strings.Contains(buildStatus.PaC.ErrMessage, "Access token is unrecognizable by GitHub"), nil
+				}, time.Minute*2, 5*time.Second).Should(BeTrue(), "failed while checking build status for component %q is correct", firstComponentName)
+			})
+
+			It(fmt.Sprintf("triggered PipelineRun is for component %s", secondComponentName), func() {
+				timeout = time.Minute * 5
+				Eventually(func() error {
+					pr, err := f.AsKubeAdmin.HasController.GetComponentPipelineRun(secondComponentName, applicationName, testNamespace, "")
+					if err != nil {
+						GinkgoWriter.Printf("PipelineRun has not been created yet for the component %s/%s\n", testNamespace, secondComponentName)
+						return err
+					}
+					if !pr.HasStarted() {
+						return fmt.Errorf("pipelinerun %s/%s hasn't started yet", pr.GetNamespace(), pr.GetName())
+					}
+					return nil
+				}, timeout, constants.PipelineRunPollingInterval).Should(Succeed(), fmt.Sprintf("timed out when waiting for the PipelineRun to start for the component %s/%s", secondComponentName, testNamespace))
+			})
+
+			It("check only one pipelinerun should be triggered", func() {
+				// Waiting for 2 minute to see if only one pipelinerun is triggered
+				Consistently(func() (bool, error) {
+					pipelineRuns, err := f.AsKubeAdmin.HasController.GetAllPipelineRunsForApplication(applicationName, testNamespace)
+					if err != nil {
+						return false, err
+					}
+					if len(pipelineRuns.Items) != 1 {
```

Comment:
What if pipeline finished earlier than 2 minutes? Should we check `len(pipelineRuns.Items) < 2`?

Reply:
Even if the pipelinerun is in completed state, `GetAllPipelineRunsForApplication` will return it and will have 2 pipelinerun count.

Reply:
I think both is `len(pipelineRuns.Items) != 1` or `len(pipelineRuns.Items) < 2` same, we can keep any one of them